src/Server/Repositories/DicomAdapterRepository.cs

```diff
@@ -110,5 +115,11 @@ public T FirstOrDefault(Func<T, bool> func)
 
             return _dicomAdapterContext.Set<T>().FirstOrDefault(func);
         }
+
+        public void Detach(T item)
+        {
+            Guard.Against.Null(item, nameof(item));
+            _dicomAdapterContext.Entry(item).State = EntityState.Detached;
```

Comment:
What if item does not exist in Entry?

Reply:
Looking at the source code to `Entry`, it looks like it'll create a new item, which is fine in our use case.